core/app/models/spree/price.rb

```diff
@@ -46,6 +46,8 @@ def price=(price)
     end
 
     def net_amount
+      return nil unless amount
```

Comment:
I don't believe it's possible for `amount` to be nil as of v3.1. (See the "Other important changes" section in the [`CHANGELOG.md`](https://github.com/solidusio/solidus/blob/main/CHANGELOG.md#solidus-310-v31-2021-09-10) entry for v3.1, or the PR that removes `nil` as a valid value: https://github.com/solidusio/solidus/pull/3987)

Which means this shouldn't be possible on recent versions of Solidus. (Assuming the task to remove any records with a `nil` amount has been run and there isn't any invalid data left in the database from before that change.)

Reply:
Oops, nevermind. I missed that we were trying to fix a `before_validation` callback on `Spree::Variant`. 

In which case I think the preferred fix should be skipping that callback instead of modifying this public method. The callback already has a conditional associated to it, so I believe it would be a less invasive change to prevent it from running in the case when a price is invalid. (And we avoid any future confusion from having a `nil` check in a method where a `nil` value is considered invalid.)